### PR TITLE
checked_yaml: improve handling of missing keys

### PR DESCRIPTION
**Pull request by kevmoo**
If a CheckedFromJsonException exception is caught for a key that is missing
in the source map, include that information in the message

Also correctly handle the case where CheckedFromJsonException.message
is `null`

Fixes https://github.com/google/json_serializable.dart/issues/816

### CODE REVIEW

**Comment by kevmoo** [comment]
@jakemac53 – related to your feedback on https://github.com/dart-lang/pubspec_parse/pull/65

File: checked_yaml/lib/checked_yaml.dart [hunk 1]
```diff
@@ -75,12 +75,18 @@ ParsedYamlException toParsedYamlException(
...
+        yamlMap,
+        innerError: exception,
+      );
+    } else if (!yamlMap.containsKey(exception.key)) {
       return ParsedYamlException(
-        exception.message!,
+        [
+          'Missing key "${exception.key}".',
+          if (exception.message != null) exception.message!,
+        ].join(' '),
```
Comment (jakemac53) [inline checked_yaml/lib/checked_yaml.dart:89]:
```suggestion
        'Missing key "${exception.key}".${exception.message ?? ''}',
```

**Review by jakemac53 — APPROVED** [review]

File: checked_yaml/lib/checked_yaml.dart (hunk 1 above)
Comment (kevmoo) [reply checked_yaml/lib/checked_yaml.dart:89]: ...but we want a space between them? Then you need to do double wrapping of string literals, etc. I think the code I wrote here is more readable. Less efficient, but it's not like this is going to be perf-critical.

Comment (natebosch) [reply checked_yaml/lib/checked_yaml.dart:89]: 😈 

```suggestion
'Missing key "${exception.key}"' + ' ${exception.message ?? ''}'.trimRight();
```

Comment (kevmoo) [reply checked_yaml/lib/checked_yaml.dart:89]: image

**Pull request merged**